src/Controls/src/Core/TypedBindingFactory.cs

```diff
@@ -0,0 +1,287 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Microsoft.Maui.Controls.Internals;
+using Microsoft.Maui.Graphics;
+
+namespace Microsoft.Maui.Controls
+{
+	internal static class TypedBindingFactory
```

Comment:
Looking at the code in here, it supports:

* ParameterExpression
* UnaryExpression
* MemberExpression
* MethodCallExpression
* BinaryExpression 

Does this need to be exhaustive yet? Should we at least put a `<summary>` comment of which expressions would work?

It seems like we'd need to implement them all if this was able to be used by customers -- which, isn't the case yet.

Reply:
I think it's perfectly OK to support only a subset and throw if the provided expression is too complex. ASP.NET uses similar trick and also implements only a limited subset of the expression tree support. For example https://github.com/dotnet/aspnetcore/blob/c2a9c205aa65af4031e0fd6471c2aa89bde095c0/src/Mvc/Mvc.ViewFeatures/src/ExpressionMetadataProvider.cs#L72C59-L72C94.

Reply:
I focused only on mirroring what customers can do with binding paths - accessing members, arrays, and indexers (as described in https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/data-binding/binding-path). From those 5 expression types, it also only supports a subset of them (for example unary operator - just `Convert`, `AsType`, method call - just special name methods `get_Indexer`, `get_Item`), but it's all that's necessary to support the expressions that are similar to the string paths in classic bindings. As @StephaneDelcroix told me, the subset of bindings I implemented is very close to one of the earlier expression-based APIs that Xamarin.Forms once had (https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Core/Binding.cs#L332).

You're definitely right that the documentation part is lacking. I'm assuming we'll need to update the official documentation if this feature goes in later, but I can definitely add that `<summary>` in this PR. Thanks for the suggestion.